packages/kbn-performance-testing-dataset-extractor/src/es_client.ts

```diff
@@ -172,9 +177,45 @@ export class ESClient {
     return await this.getTransactions<TransactionDocument>(queryFilters);
   }
 
-  async getSpans(transactionId: string) {
-    const filters = [{ field: 'parent.id', value: transactionId }];
-    const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-    return await this.getTransactions<SpanDocument>(queryFilters);
+  getMsearchRequestItem = (queryFilters: QueryDslQueryContainer[]): MsearchRequestItem => {
+    return {
+      query: {
+        bool: {
+          filter: [
+            {
+              bool: {
+                filter: queryFilters,
+              },
+            },
+          ],
+        },
+      },
+    };
+  };
+
+  async getSpans(transactionIds: string[]) {
+    const searches = new Array<MsearchRequestItem>();
+
+    for (const transactionId of transactionIds) {
+      const filters = [{ field: 'parent.id', value: transactionId }];
+      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
+      const requestItem = this.getMsearchRequestItem(queryFilters);
+      searches.push({}, requestItem);
+    }
+    this.log.debug(`Msearch request: ${JSON.stringify(searches)}`);
+    const result = await this.client.msearch<SpanDocument>({
+      searches,
+    });
+    this.log.debug(`Msearch result: ${JSON.stringify(result)}`);
+    const hits = result.responses
+      .map((response) => response as SearchResponse<SpanDocument>)
+      .filter(
+        (response) => response.hits.total && (response.hits.total as SearchTotalHits).value > 0
+      )
+      .map((response) => response.hits.hits)
+      .reduce((acc, value) => {
+        return acc.concat(value);
+      });
+    return hits;
```

Comment:
I recommend avoiding type casting with `as`, and using `flatMap()`
```suggestion
    this.log.debug(`Msearch result: ${JSON.stringify(result)}`);
    return result.responses.flatMap((response) => {
      if ('error' in response) {
        throw new Error(`msearch failure: ${JSON.stringify(response.error)}`);
      } else if (response.hits.hits.length > 0) {
        return response.hits.hits;
      } else {
        return [];
      }
    });
```